### PR TITLE
fix: make pubsub.unsubscribe async and alter pubsub.subscribe signature

### PR DESCRIPTION
**Pull request by diasdavid**
Fix the flaky tests and API.

### CODE REVIEW

**Comment by diasdavid** [comment]
Missing:

- [ ] Update tests
- [ ] Update js-ipfs-api
- [ ] Update js-ipfs

I want to land this before the next release of js-ipfs

**Comment by alanshaw** [comment]
Ha, nice. I had to make `unsubscribe` async in `ipfs-postmsg-proxy` https://github.com/tableflip/ipfs-postmsg-proxy#pubsub-unsubscribe as there's no way to sync over `postMessage` so the spec is now in line with my implementation :D

File: SPEC/PUBSUB.md [hunk 1]
```diff
@@ -7,13 +7,13 @@ pubsub API
...
 
-- `topic: string`
-- `options: Object` - (Optional), might contain the following properties:
-  - `discover`: type: Boolean - Will use the DHT to find other peers.
-- `handler: (msg) => ()` - Event handler which will be called with a message object everytime one is received. The `msg` has the format `{from: string, seqno: Buffer, data: Buffer, topicIDs: Array<string>}`.
-- `callback: (Error) => ()` (Optional) Called once the subscription is established.
+- `options: Object`: Object containing the following properties:
+  - `topic: string`
+  - `discover: Boolean` - Will use the DHT to find other peers.
+  - `handler: (msg) => ()` - Event handler which will be called with a message object everytime one is received. The `msg` has the format `{from: string, seqno: Buffer, data: Buffer, topicIDs: Array<string>}`.
```
Comment (VictorBjelkholm) [inline SPEC/PUBSUB.md]: The handler moves into the options object? That's a bit of a weird pattern, much nicer when it was separated.

Comment (diasdavid) [reply SPEC/PUBSUB.md]: I believe your proposal of:

```
ipfs.pubsub.subscribe(topic, handler, options, callback)
ipfs.pubsub.unsubscribe(topic, handler, callback)
```

Pretty much solves the main issue. I like it!

**Comment by diasdavid** [comment]
@alanshaw wanna take this API, spec and tests change from me?

**Comment by alanshaw** [comment]
Yes

**Comment by alanshaw** [comment]
@diasdavid I think this and https://github.com/ipfs/js-ipfs/pull/1348 and https://github.com/ipfs/js-ipfs-api/pull/760 are good for a review now 🚀

**Comment by diasdavid** [comment]
@alanshaw feel empowered to merge and release this one :) (as minor).

**Comment by alanshaw** [comment]
I am need 1 approving review

**Pull request merged**